### PR TITLE
Use RAII MainWindow toolbarWidgets

### PR DESCRIPTION
**Pull request by jhilmer**
Also changed to RAII in ToolbarDragDropHandler as it was using the toolbarWidgets of MainWindow.

### CODE REVIEW

File: src/core/gui/dialog/toolbarCustomize/ToolbarDragDropHandler.cpp [hunk 1]
```diff
@@ -16,37 +16,28 @@
...
-    this->toolbars[len] = nullptr;
+    this->toolbars.clear();
+    this->toolbars.reserve(len);
 
-    for (int i = 0; i < len; i++) {
-        GtkWidget* w = widgets[i];
-        this->toolbars[i] = new ToolbarAdapter(w, win->getToolbarName(GTK_TOOLBAR(w)),
-                                               control->getWindow()->getToolMenuHandler(), control->getWindow());
+    for (size_t i = 0; i < len; i++) {
+        GtkWidget* w = win->getToolbarWidget(i);
```
Comment (bhennion) [inline src/core/gui/dialog/toolbarCustomize/ToolbarDragDropHandler.cpp:29]: Can we get a const reference for the `std::array` and iterate on it directly here?

Comment (jhilmer) [reply src/core/gui/dialog/toolbarCustomize/ToolbarDragDropHandler.cpp:29]: Yes, and then we don't have a need for the above `getToolbarCount`

**Review by bhennion — APPROVED** [review]
LGTM! Merging in 24 hours

**Comment by jhilmer** [comment]
Squashed the above, so it is ready to merge

**Pull request merged**